openwisp_notifications/api/views.py

```diff
@@ -145,6 +145,16 @@ def get(self, request, *args, **kwargs):
 class NotificationSettingView(BaseNotificationSettingView, RetrieveUpdateAPIView):
     lookup_field = "pk"
 
+    def get_object(self):
+        if self.kwargs["pk"] == "null":
+            # Handle global setting (organization=None, type=None)
+            return get_object_or_404(
+                self.get_queryset(),
+                organization__isnull=True,
+                type__isnull=True
+            )
+        return super().get_object()
+
```

Comment:
Can you explain the logic behind this change? Why do we return an object when pk is None?

The current state of code already handles changing of global notification settings. We only need to handle the corner case where user re-attempts to disable the already disabled global notification setting.